packages/x-charts/src/internals/plugins/corePlugins/useChartDimensions/useChartDimensions.ts

```diff
@@ -197,18 +198,51 @@ export const useChartDimensions: ChartPlugin<UseChartDimensionsSignature> = ({
     [drawingArea.height, drawingArea.top],
   );
   const isPointInside = React.useCallback(
-    (x: number, y: number, targetElement?: Element) => {
+    (x: number, y: number, targetElement?: Element | EventTarget | null) => {
       // For element allowed to overflow, wrapping them in <g data-drawing-container /> make them fully part of the drawing area.
-      if (targetElement && targetElement.closest('[data-drawing-container]')) {
+      if (
+        targetElement &&
+        'closest' in targetElement &&
+        targetElement.closest('[data-drawing-container]')
+      ) {
         return true;
       }
 
       return isXInside(x) && isYInside(y);
     },
     [isXInside, isYInside],
   );
+  const isElementInside = React.useCallback(
+    (element: Element | EventTarget | null | undefined, partial = false) => {
+      const svgElement = svgRef.current;
+      if (!element || !(element instanceof Element) || !svgElement) {
+        return false;
+      }
+      // For element allowed to overflow, wrapping them in <g data-drawing-container /> make them fully part of the drawing area.
+      if (element && element.closest('[data-drawing-container]')) {
+        return true;
+      }
+
+      const rect = element.getBoundingClientRect();
```

Comment:
We should be careful of calling `getBoundingClientRect` too often, as it might cause performance problems ([more info](https://webperf.tips/tip/layout-thrashing/)). 

Nothing to do here for now, but we should be aware in case it becomes a problem.